homeassistant/helpers/script_variables.py

```diff
@@ -65,6 +60,216 @@ def async_render(
 
         return rendered_variables
 
+    @callback
+    def async_simple_render(self, run_variables: Mapping[str, Any]) -> dict[str, Any]:
+        """Render script variables.
+
+        Simply renders the variables, the run variables are not included in the result.
+        The run variables are used to compute the rendered variable values.
+        The rendering happens one at a time, with previous results influencing the next.
+        """
+        if self._has_template is None:
+            self._has_template = template.is_complex(self.variables)
+
+        if not self._has_template:
+            return self.variables
+
+        run_variables = dict(run_variables)
+        rendered_variables = {}
+
+        for key, value in self.variables.items():
+            rendered_variable = template.render_complex(value, run_variables)
+            rendered_variables[key] = rendered_variable
+            run_variables[key] = rendered_variable
+
+        return rendered_variables
+
     def as_dict(self) -> dict[str, Any]:
         """Return dict version of this class."""
         return self.variables
+
+
+@dataclass
+class _ParallelStore:
+    """Store used in each parallel sequence."""
+
+    # `special` is for variables that need special treatment in parallel sequences.
+    # Currently the only such variable is `wait`.
+    special: dict[str, Any] = field(default_factory=dict)
```

Comment:
"Special" really doesn't explain anything, what's special about the special variables?